src/data-access/apiAccess.ts

```diff
@@ -0,0 +1,163 @@
+        throw error;
+    }
+}
+
+export async function getMovieTrailers(show_id: string | undefined): Promise<MovieTrailersDTO[]> {
+    try {
+        const trailerURL = `${baseUrl}movie/${show_id}/videos`;
+        const response = await axiosInstance.get(trailerURL, {
+            params: {
+                api_key: API_KEY,
+            },
+        });
+        const movieTrailersResponse: MovieTrailersDTO[] = response.data.results;
+        return movieTrailersResponse;
+    } catch (error) {
+        console.error('Error fetching movie trailers:', error);
+        throw error;
+    }
+}
+
+export async function getRandomMovie(randomPage: number): Promise<RandomMovieDTO> {
+    try {
+        const randomURL = `${randomMovieURL}${randomPage}`;
+        const response = await axiosInstance.get(randomURL);
+        const randomMovieResponse: RandomMovieDTO = response.data;
+        return randomMovieResponse;
+    } catch (error) {
+        console.error('Error fetching random movie:', error);
+        throw error;
+    }
+}
+
+export async function getPersonDetails(personId: string | undefined): Promise<PersonDetailsDTO> {
+    try {
+        const personDetailsURL = `${baseUrl}person/${personId}`;
+        const response = await axiosInstance.get(personDetailsURL, {
+            params: {
+                api_key: API_KEY,
+            },
+        });
+        const personDetailsResponse: PersonDetailsDTO = response.data;
+        return personDetailsResponse;
+    } catch (error) {
+        console.error('Error fetching person details:', error);
+        throw error;
+    }
+}
+
+export async function getPersonPopMovies(personId: string | undefined): Promise<PersonPopMoviesDTO> {
+    try {
+        const personPopMoviesURL = `${baseUrl}person/${personId}/movie_credits`;
+        const response = await axiosInstance.get(personPopMoviesURL, {
+            params: {
+                api_key: API_KEY,
+            },
+        });
+        const personPopMoviesResponse: PersonPopMoviesDTO = response.data;
+        return personPopMoviesResponse;
+    } catch (error) {
+        console.error('Error fetching pop movies:', error);
+        throw error;
+    }
+}
+
+export async function getSearchPerson(page: number, query: string): Promise<SearchPersonDTO[]> {
+    try {
+        const response = await axiosInstance.get(personSearchUrl, {
+            params: {
+                page: page,
+                query: query,
+                api_key: apiKey
+            }
+        });
+        const personSearchResponse: SearchPersonDTO[] = response.data.results;
+        return personSearchResponse;
+    } catch (error) {
+        console.error('Error fetching movie trailers:', error);
+        throw error;
+    }
+}
+
+export async function getSearchMovies(page: number, query: string, genreID: string | null): Promise<SearchMovieDTO[]> {
+    try {
+        const searchUrl = `${baseUrl}search/movie?api_key=${apiKey}&query=${query}`;
```

Comment:
Discover and search looks two different api endpoints, I don't think that we should mix them here. 

Let's separate them and create two functions and then in the application layer we can decide which one we should call. Not here